sdk/scheduler/src/main/java/com/mesosphere/sdk/state/PersistentOperationRecorder.java

```diff
@@ -34,7 +36,13 @@ public void record(Operation operation, Offer offer) throws Exception {
     private void recordTasks(List<Protos.TaskInfo> taskInfos) throws StateStoreException {
         logger.info(String.format("Recording %d updated TaskInfos/TaskStatuses:", taskInfos.size()));
         List<Protos.TaskStatus> taskStatuses = new ArrayList<>();
-        for (Protos.TaskInfo taskInfo : taskInfos) {
+        Collection<Protos.TaskInfo> unpackedTaskInfos;
+        try {
+            unpackedTaskInfos = CommonTaskUtils.unpackTaskInfos(taskInfos);
```

Comment:
I see the point that you are making. But, we cannot do `operation instanceof LaunchOfferRecommendation` as `LaunchOfferRecommendation` IS-NOT-A `Operation` but IS-A `OfferRecommendation` which HAS-A `Operation`. Let me see what we can do here to simplify this further.